simphony/testing/abc_check_engine.py

```diff
@@ -35,46 +45,48 @@ def create_dataset_items(self):
         """
 
     @abc.abstractmethod
-    def check_emtpy_dataset(self, ds):
-        """ Check if a dataset is empty
-        """
-
-    @abc.abstractmethod
     def check_instance_of_dataset(self, ds):
         """ Check if a dataset is instance of a class
         """
 
-    @abc.abstractmethod
-    def check_is_usable_dataset(self, ds, items):
-        """ Check if a dataset is usable
-        """
+    operation_mapping = {
+        'compare datasets': 'none',
+        'add item': 'none'}
+
+    def compare_operation(self, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
```

Comment:
I want to compare datasets here:

https://github.com/simphony/simphony-common/blob/873fd3378826024d3e2993836f5933cca9be0f5a/simphony/testing/abc_check_engine.py#L79

but I cannot use ```assertEqual``` as _reference_ is an instance of let's say ```Particles``` and the container retrieved _ds_ is an instance of ```H5Particles```.

So either I call ```sys.modules``` or I import all ```simphony.testing.utils``` and then use ```utils``` as the module.